Update CONTRIBUTING.md (#49221)
elastic/elasticsearch: CONTRIBUTING.md

For Eclipse, the minimum version that we support is [4.13][eclipse]. 
For IntelliJ, the minimum version that we support is [IntelliJ 2017.2][intellij].

> [Docker](https://docs.docker.com/install/) is a necessary dependency for building Elasticsearch. You can run Elasticsearch without building all the artifacts with:

Reviewer: Sorry if I'm overly pedantic about this but can we be a bit more accurate and say docker "is required for building _some_ Elasticsearch artifacts and executing certain test suites."

For Eclipse, the minimum version that we support is [4.13][eclipse]. 
For IntelliJ, the minimum version that we support is [IntelliJ 2017.2][intellij].

> [Docker](https://docs.docker.com/install/) is a necessary dependency for building elasticsearch. Make sure docker is installed prior to running an instance of elasticsearch from source code.

Reviewer 2: @ejmiers thanks for confirming!

Regarding the change, how about the following?

```suggestion
[Docker](https://docs.docker.com/install/) is a necessary dependency for building Elasticsearch. You can run Elasticsearch without building all the artifacts with:

    ./gradlew :run
```
